editor/import/3d/resource_importer_scene.cpp

```diff
@@ -1520,6 +1533,21 @@ Node *ResourceImporterScene::_post_fix_node(Node *p_node, Node *p_root, HashMap<
 					Ref<Material> mat = m->get_surface_material(i);
 					if (mat.is_valid()) {
 						String mat_id = mat->get_meta("import_id", mat->get_name());
+						if (extract_mat != 0) {
+							String ext = (p_options.has("materials/extract_format") && (int)p_options["materials/extract_format"] == 1) ? ".res" : ".tres";
+							String path = spath.path_join(mat_id.validate_filename() + ext);
+							Dictionary matdata = p_material_data[mat_id];
+							matdata["use_external/enabled"] = true;
+							matdata["use_external/path"] = path;
```

Comment:
use_external/path should be the uid:// path and fallback_path should be res://

the fact that we now have effectively duplicate code (with SceneImportSettingsDialoh) for extracting materials has me a little worried that there will be a divergence in behavior between the two (such as this uid assignment code as an  example)

Reply:
The only duplicate code is `ResourceSaver::save(md.material, path);` the rest can't be extracted since `SceneImportSettingsDialog` and `ResourceImporterScene` are using different types.

Reply:
Added "UID" and `fallback_path` handing.